components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/user/core/store/CacheBackedIdentityStore.java

```diff
@@ -86,7 +87,8 @@ public void init(DomainManager domainManager, Map<String, IdentityStoreConnector
     }
 
     @Override
-    public User getUser(String username) throws IdentityStoreException, UserNotFoundException {
+    public User getUser(String username)
+            throws IdentityStoreException, UserNotFoundException, DomainException {
```

Comment:
Domain exception should be wrapped from IdentityStoreException since the caller does not have any value in knowing them differently.